lib/chunky_png/chunk.rb

```diff
@@ -319,6 +319,45 @@ def content
       end
     end
 
+    # The Physical (pHYs) chunk specifies the intended pixel size or aspect
+    # ratio for display of the image.
+    #
+    # http://www.libpng.org/pub/png/spec/1.2/PNG-Chunks.html#C.pHYs
+    #
+    class Physical < Generic
+      attr_accessor :ppux, :ppuy, :unit
+
+      def initialize(ppux, ppuy, unit=:unknown)
+        raise ArgumentError, 'unit must be either :meters or :unknown' unless [:meters, :unknown].member?(unit)
+        super('pHYs')
+        @ppux, @ppuy, @unit = ppux, ppuy, unit
+      end
+
+      def dpix
+        ppux * INCHES_PER_METER if unit == :meters
+      end
+
+      def dpiy
+        ppuy * INCHES_PER_METER if unit == :meters
```

Comment:
I'd prefer raising an exception. It will give nice error messages in case somebody forgets to check whether the unit used is not unknown.